qiskit_ibm_runtime/utils/queueinfo.py

```diff
@@ -0,0 +1,175 @@
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Queue information for a job."""
+
+import warnings
+from datetime import datetime
+from typing import Any, Optional, Union, Dict
+import dateutil.parser
+
+from qiskit_ibm_provider.utils.utils import api_status_to_job_status
+
+from ..utils import utc_to_local, duration_difference
+
+
+class QueueInfo:
+    """Queue information for a job."""
+
+    _data = {}  # type: Dict
+
+    def __init__(
+        self,
+        position_in_queue: Optional[int] = None,
+        status: Optional[str] = None,
+        estimated_start_time: Optional[Union[str, datetime]] = None,
+        estimated_completion_time: Optional[Union[str, datetime]] = None,
+        hub_priority: Optional[float] = None,
+        group_priority: Optional[float] = None,
+        project_priority: Optional[float] = None,
+        job_id: Optional[str] = None,
+        **kwargs: Any
+    ) -> None:
+        """QueueInfo constructor.
+
+        Args:
+            position: Position in the queue.
+            status: Job status.
+            estimated_start_time: Estimated start time for the job, in UTC.
+            estimated_complete_time: Estimated complete time for the job, in UTC.
+            hub_priority: Dynamic priority for the hub.
+            group_priority: Dynamic priority for the group.
+            project_priority: Dynamic priority for the project.
+            job_id: Job ID.
+            kwargs: Additional attributes.
+        """
+        self.position = int(position_in_queue) if position_in_queue else None
+        self._status = status
+        if isinstance(estimated_start_time, str):
+            estimated_start_time = dateutil.parser.isoparse(estimated_start_time)
+        if isinstance(estimated_completion_time, str):
+            estimated_completion_time = dateutil.parser.isoparse(estimated_completion_time)
+        self._estimated_start_time_utc = estimated_start_time
+        self._estimated_complete_time_utc = estimated_completion_time
+        self.hub_priority = hub_priority
+        self.group_priority = group_priority
+        self.project_priority = project_priority
+        self.job_id = job_id
+
+        self._data = kwargs
+
+    def __repr__(self) -> str:
+        """Return the string representation of ``QueueInfo``.
+
+        Note:
+            The estimated start and end time are displayed in local time
+            for convenience.
+
+        Returns:
+            A string representation of ``QueueInfo``.
+
+        Raises:
+            TypeError: If the `estimated_start_time` or `estimated_end_time`
+                value is not valid.
+        """
+        status = (
+            api_status_to_job_status(self._status).name
```

Comment:
If the job status is passed in, I don't think `api_status_to_job_status` will be necessary because the status should already be in the right format

Reply:
okay. removed.